media/js/base/mozilla-fxa-form.js

```diff
@@ -102,6 +100,55 @@ if (typeof window.Mozilla === 'undefined') {
         });
     };
 
+
+    /**
+     * Intercept event handler for FxA forms, lets the browser drive the FxA Flow using
+     * the `showFirefoxAccounts` UITour API. Attaches several UTM parameters from the current page
+     * that will be forwarded to the browser and later on to FxA services.
+     * @param event {Event}
+     * @private
+     */
+    FxaForm.interceptFxANavigation = function(event) {
+        event.preventDefault();
+        var extraURLParams = FxaForm.getUTMParams();
+
+        var entrypointInput = document.getElementById('fxa-email-form-entrypoint');
+        var entrypointExp = document.getElementById('fxa-email-form-entrypoint-experiment');
+        var entrypointVar = document.getElementById('fxa-email-form-entrypoint-variation');
+        var entrypoint = null;
+        if (entrypointInput && entrypointInput.value) {
+            entrypoint = entrypointInput.value;
+        }
+        if (entrypointExp && entrypointExp.value) {
+            extraURLParams['entrypoint_experiment'] = entrypointExp.value;
+        }
+        if (entrypointVar && entrypointVar.value) {
+            extraURLParams['entrypoint_variation'] = entrypointVar.value;
+        }
+
+        var email = document.getElementById('fxa-email-field');
+        if (email) {
+            email = email.value;
+        }
+
+        var formElem = document.getElementById('fxa-email-form');
+        if (formElem) {
+            var deviceId = formElem.querySelector('[name="device_id"]');
+            var flowId = formElem.querySelector('[name="flow_id"]');
+            var flowBeginTime = formElem.querySelector('[name="flow_begin_time"]');
+            if (deviceId && deviceId.value) {
+                extraURLParams['device_id'] = deviceId.value;
+            }
+            if (flowId && flowId.value) {
+                extraURLParams['flow_id'] = flowId.value;
+            }
+            if (flowBeginTime && flowBeginTime.value) {
+                extraURLParams['flow_begin_time'] = parseInt(flowBeginTime.value, 10);
+            }
+        }
+        return Mozilla.UITour.showFirefoxAccounts(extraURLParams, entrypoint, email);
```

Comment:
>  Similarly, do we need a sane default for entrypoint which isn't null? What would happen if the form doesn't contain an email field? etc

For `entrypoint` the browser checks for it. It will default to `uitour` if nothing is reported.
For `email` the browser checks for that and has a fork in the logic: https://searchfox.org/mozilla-central/source/browser/components/uitour/UITour.jsm#531